Tribler/community/doubleentry/conversion.py

```diff
@@ -0,0 +1,148 @@
+from Tribler.Core.Utilities.encoding import encode, decode
+
+from Tribler.dispersy.conversion import BinaryConversion
+from Tribler.dispersy.message import DropPacket
+
+
+class DoubleEntryConversion(BinaryConversion):
+    """
+    Class that handles all encoding and decoding of DoubleEntry messages.
+    """
+
+    def __init__(self, community):
+        super(DoubleEntryConversion, self).__init__(community, "\x01")
+        from Tribler.community.doubleentry.community import SIGNATURE_REQUEST
+        from Tribler.community.doubleentry.community import SIGNATURE_RESPONSE
+
+        # Define Request Signature.
+        self.define_meta_message(chr(1), community.get_meta_message(SIGNATURE_REQUEST),
+                                 self._encode_signature_request, self._decode_signature_request)
+        self.define_meta_message(chr(2), community.get_meta_message(SIGNATURE_RESPONSE),
+                                 self._encode_signature_response, self._decode_signature_response)
+
+    def _encode_signature_request(self, message):
+        """
+        Encode a signature_request message.
+        :param message: The message to be encoded.
+        :return: encoded signature request message.
+        """
+        payload = message.payload
+        return encode((payload.up, payload.down, payload.total_up, payload.total_down,
```

Comment:
If you like to do some more work, you could switch to using structs. Have a look at the other communities for an example. They're a bit more efficient + you can force the type of the parameters.